### ISSUE TITLE
Loading slow the home screen when returning

### ISSUE THREAD

**Issue by DreVla** [8bitPit/Niagara-Issues#2658]
### Describe the bug

When in an app and then returning to niagara home by using the home button, it takes quite a while to load all the widgets & icons.

### Steps to reproduce

1. Go to any app
2. Press home button / Do home gesture
3. Loading is slow

### Does the issue also occur with other 3rd-party launchers?

No

### Other 3rd-party launcher(s) tested

Tested on default launcher and Nova launcher and the issue is not reproducible.

### Device name

OnePlus 8T

### Android version

Android 14

### Screenshots / Screen recordings

https://github.com/8bitPit/Niagara-Issues/assets/32488956/1e1deff9-83f1-4a67-98c0-0b1740923120

**Comment by DreVla** [comment]
Additional note - This doesn't happen when returning home by pressing back button / doing back gesture multiple times

**Comment by TheKineticz** [comment]
If you revoke notification access, does this issue persist?

**Comment by DreVla** [comment]
> If you revoke notification access, does this issue persist?

Yes, disabling notification access doesn't change anything, the issue still persists.

**Comment by jwalkingjon** [comment]
Same issue on my oneplus open, swipping up home causes a 1-2 second lag but backing out of the app is fine.

**Comment by NiagaraSupport** [comment]
Thank you all for the reports.
We'll get back to you once we know more or the bug has been fixed.

**Comment by coolrack** [comment]
> Thank you all for the reports. This bug should be fixed with update 1.11.6. Please update us if the problem persists.

hey, been an issue for a while, on ColorOS 14. Same issue depicted in video from OP.

**Comment by dpihlaja** [comment]
This is still present it seems on 1.12.3, on latest OxygenOS 14.0.0.802.

Swiping upwards presents the lag when closing an app, but hitting the back button does not.

Also from the recent apps screen, swiping up presents the lag but swiping back does not.

**Comment by jwalkingjon** [comment]
Still an issue for me on 1.12.3 latest Oxygen OS version as well

**Comment by AllDragonsAreDead** [comment]
I have exactly same issue with OnePlus Nord 4. On Pixel 6a everything was working well. It seems OnePlus related.

**Comment by rocuevas9511** [comment]
Reporting this happening on color os 15
CPH2659_15.0.0.301(EX01)
, oppo find x8 pro.

Niagara v 1.13.1

**Comment by outman-2024** [comment]
Same issue

**Comment by teegles** [comment]
Same issue here on oxygenos CPH2551_15.0.0.210(EX01) on OnePlus Open

Niagara Launcher 1.13.1

Appreciate you all!

**Comment by DovieW** [comment]
Found a better solution!

Enable 3 button nav, and then run this ADB command:
adb shell cmd overlay enable com.android.internal.systemui.navbar.gestural

This seems to give you the gesture bar at the bottom but it's broken, it doesn't do anything. So now you can use other Gesture apps like Vivid Navigation Gestures.

**Comment by dhruvil007** [comment]
Hi, any updates on when a fix may be released? I'm seeing this on my OnePlus 13, Oxygen OS 13.

**Comment by DovieW** [comment]
well it's specifically an issue with oneplus it seems. i dont have this issue on pixel and samsung. i returned my oneplus. too many issues

**Comment by DovieW** [comment]
I take it back. I have the s25 Ultra with OneUI 7 and it seems the issue is still there albeit slightly different. It appears that everything has loaded (it's not blank like it was on the Oneplus), but the screen is unresponsive. Almost as if the issue is the same but in the Samsung's case, you can at least see the home screen so it feels less obnoxious

**Comment by ChocoFleece** [comment]
The problem still exists in OnePlus 13

**Comment by revfast** [comment]
Im also having this same issue on Oneplus 13

**Comment by chartzngrafs** [comment]
Can confirm on OnePlus 13R with Oxy15.0. This issue only exists in Gesture mode. Totally fine if I have it set to button navigation. Any ETR on this?

**Comment by chartzngrafs** [comment]
Can confirm on OnePlus 13R with Oxy15.0. This issue only exists in Gesture mode. Totally fine if I have it set to button navigation. Any ETR on this? Additionally, I thought removing interface animations in the Accessibility option could potentially resolve, it does not.

**Comment by Kairosen** [comment]
I'm on OnePlus 13, Android 15. Same issue for me, makes it unusable... I really hope this gets fixed, I can't live without Niagara.

**Comment by Kubikunde** [comment]
Same issue on OnePlus 13 & OnePlus Pad 3 😭 (Nova pro version)
On my Tablet I switched back to stock launcher but this is no option for me on my OnePlus 13

**Comment by RetroPandroo** [comment]
+1 for OnePlus 13 😢  CPH2653_15.0.0.831(EX01)

Didn't used to have this problem on my Nothing Phone 1. 

Tried messing with animations settings (standard Vs advanced, speed scaling) to no avail. 
Also, I've been slowly setting up my OnePlus and transferring everything over, and I only noticed this "1-second-blank-homescreen" issue towards the end of my "logging into every app spree" 🤔

**Comment by snivets** [comment]
Another OnePlus 13 here with the same issue, funnily enough also coming from a Nothing. It seems to also happen on my OnePlus Pad.

**Comment by aintnodev** [comment]
Realme user here, with same issue.
Was using vxxx earlier that didn't even let me use navi

**Comment by aintnodev** [comment]
Realme user here, was using `v15.4.18` earlier, but navigation gesture was broken. Sideloading latest `v15.8.19` of OnePlus System launcher fixed that issue, but home screen delay persists.

**Comment by dpihlaja** [comment]
@NiagaraSupport has migrated off of github FYI

https://github.com/NiagaraLauncher/Niagara-Issues?tab=readme-ov-file#we-have-migrated-away-from-github

This seems to be the new issue thread
https://feedback.niagaralauncher.com/p/missing-animation-2

**Comment by teegles** [comment]
This is also not a Niagara issue, it's an issue with OnePlus because every
3rd party launcher experiences this. Unfortunately this has happened for
quite a while and there has been nothing from oneplus about fixing it.

**Comment by TheBestPessimist** [comment]
@NiagaraSupport if you migrated from Github, please disable issues of this repo or archive the repo completely. Otherwise people will write here in hope of support, but they'll only be yelling at the empty sky.

**Comment by Quicksoapy** [comment]
+1 on oneplus pad 3 with android 16
